src/character.cpp

```diff
@@ -2088,21 +2097,27 @@ hp_part Character::body_window( const std::string &menu_header,
     };
 
     const auto hp_str = [precise]( const int hp, const int maximal_hp ) -> std::string {
-        if( hp <= 0 ) {
-            return "-----";
-        } else if( precise ) {
+        if( hp <= 0 )
+        {
+            return "==%==";
+        } else if( precise )
+        {
             return string_format( "%d", hp );
-        } else {
-            return string_format( "%-5s", get_hp_bar( hp, maximal_hp, false ).first );
+        } else
+        {
+            std::string h_bar = get_hp_bar( hp, maximal_hp, false ).first;
+            nc_color h_bar_col = get_hp_bar( hp, maximal_hp, false ).second;
+
+            return string_format( "%s%s", tag_colored_string( h_bar, h_bar_col ),
```

Comment:
Why are you using `string_format` here? This looks like a simple string concatenation.